src/core/handlers/base.ts

```diff
@@ -279,8 +279,15 @@ function makeHandlerContext(
   return { context, results };
 }
 
+// return cell language handler only
 export function languages(): string[] {
-  return Object.keys(handlers);
+  const cellLanguage = [];
+  for (const [k, v] of Object.entries(handlers)) {
+    if (v.type === "cell") {
+      cellLanguage.push(k);
+    }
+  }
+  return cellLanguage;
 }
```

Comment:
@cscheid this is my proposal for the language cell issue. My understanding is that `languages()` should return only cell type handler and not others. We are using `languages()` in several places in the code, but it seems that this is what is expected. Also considering your work on https://github.com/quarto-dev/quarto-cli/commit/7c215a56e50f83d943fb35dbaa109d4374492d4f

Reply:
@cscheid if you can confirm me this is good before we merge. Thank you

Reply:
Yes, this looks exactly right, thanks! (just for contrast, I'd have written it as `Object.entries(handlers).filter(([k, v]) => v.type === "cell")` but your code looks 100% correct.)